src/main/java/com/frostwire/jlibtorrent/TorrentStats.java

```diff
@@ -0,0 +1,150 @@
+    private final Queue<Integer> downloadRate = new LinkedList<>();
+    private final Queue<Integer> uploadRate = new LinkedList<>();
+    private long tStart; //for collecting sampling interval time
+
+    public enum Metric {
+        UploadRate,
+        DownloadRate
+    }
+
+    public TorrentStats(final SessionManager sessionManager, TorrentHandle torrentHandle, long samplingIntervalInMs, long maxHistoryInMs) {
+        this.sessionManager = sessionManager;
+        this.torrentHandle = torrentHandle;
+        this.samplingIntervalInMs = samplingIntervalInMs;
+        this.tStart = System.currentTimeMillis();
+        this.MAX_SAMPLES = maxHistoryInMs / samplingIntervalInMs;
+        startAlertListener();
+    }
+
+    /**
+     * It will start the alert event listener, as soon as the sampling interval
+     * meets the requested time it will save the values on queues. If the queues
+     * reach max size denoted by the maxHistory time, they will poll() the head
+     * element, in our case the oldest inserted.
+     */
+    public void startAlertListener() {
+        sessionManager.addListener(new AlertListener() {
+            @Override
+            public int[] types() {
+                return STATS;
+            }
+
+            @Override
+            public void alert(Alert<?> alert) {
+
+                if (!(alert instanceof TorrentAlert<?>)) {
+                    return;
+                }
+                //if not eq to the torrentHandle return.
+                if (!((TorrentAlert<?>) alert).handle().swig().op_eq(torrentHandle.swig())) {
+                    return;
+                }
+
+                //if !paused && !finished, it could be paused during the download
+                if (!torrentHandle.status().isFinished() && !torrentHandle.status().isPaused()) {
+
+                    //only true if samplingIntervalInMs
+                    if ((System.currentTimeMillis() - tStart) >= samplingIntervalInMs) {
+                        tStart = System.currentTimeMillis();
+
+                        //add stats
+                        downloadRate.add(torrentHandle.status().downloadRate());
+                        uploadRate.add(torrentHandle.status().uploadRate());
+
+                        //if the sampling # exceeded the limit, remove. It keeps max size.
+                        while (MAX_SAMPLES < downloadRate.size()) {
+                            downloadRate.poll();
+                            uploadRate.poll();
+                        }
+                    }
+
+                }
+            }
+        });
+    }
+
+    /**
+     * Will return all available items on the queue, depending on the type selected
+     *
+     * @param type  type of speed (bytes/sec) data to retrieved
+     * @param limit optional, to limit the result set
+     * @return all the elements tracked by the event listener limited by the maxHistory
+     */
+
+    public int[] get(Metric type, Integer... limit) {
```

Comment:
I truly do not understand why this method takes an optional array of integers as a limit.

If you want to limit the result set, why not pass a single `int` ?

like

```java
/**
 * ...
 * @param limit, if -1, return the entire result set, otherwise return min(length_of_result_set, limit)
public int[] get(Metric type, int limit) { ...
```

Right now, this method could be used like this, and then I don't understand what you mean:
```java
get(Metric.UploadRate, 10, 12, 24, 12); // what happens if you do this? I see you're only asking for the first element any ways. I'm confused.
```